### PR TITLE
fix: use epic id instead of iid for epic notes

### LINKED ISSUE #2234: Epic notes list returns 404 error
Tried using epic.notes.list() but it's returning 404 error, this matches the behavior described in https://docs.gitlab.com/ee/api/notes.html#list-all-epic-notes when **epic IID** is used.

So guessing these need to change to epic ID instead?
https://github.com/python-gitlab/python-gitlab/blob/3f91d2455657c570851f625c0d80c685d9393cbb/gitlab/v4/objects/notes.py#L52
https://github.com/python-gitlab/python-gitlab/blob/3f91d2455657c570851f625c0d80c685d9393cbb/gitlab/v4/objects/notes.py#L54

### PR DESCRIPTION
**Pull request by Shreya-7**
fixes #2234
Corresponding GitLab issue: https://gitlab.com/gitlab-org/gitlab/-/issues/351656 

This PR also fixes the same issue related to epic discussion notes, GitLab API ref: https://docs.gitlab.com/ee/api/discussions.html#epics

### CODE REVIEW

**Comment by Shreya-7** [comment]
@nejch can you review this PR?

**Review by nejch — CHANGES_REQUESTED** [review]
Thanks a lot, looking great! just one little note on the testing, which is mostly an issue with our legacy testing.

File: tests/functional/ee-test.py [hunk 1]
```diff
@@ -154,5 +155,11 @@ def end_log():
 for ei in epic.issues.list():
     ei.delete()
 
+# epic notes
+assert not epic.notes.list()
+for i in EPIC_NOTES:
+    epic.notes.create({"body": i})
+assert len(EPIC_NOTES == len(epic.notes.list()))
```
[the remark this answers is not in the document]
Comment (nejch) [reply tests/functional/ee-test.py:162]: Actually, I don't want to block you here. We can refactor this legacy code in a separate PR.

**Review by nejch — APPROVED** [review]

File: tests/functional/ee-test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Shreya-7) [reply tests/functional/ee-test.py:162]: @nejch thanks for the merge! I’d be happy to take a shot at this as a separate issue - do we have  something tracking this?

[the remark this answers is not in the document]
Comment (nejch) [reply tests/functional/ee-test.py:162]: @Shreya-7 ah nice! Well, I already started https://github.com/python-gitlab/python-gitlab/pull/2317 for this, it was a bit more involved in the end as I started migrating all the legacy stuff.

But if you're interested (especially if you're doing hacktoberfest currently, we have the label enabled), you can take a look at issues with `help wanted` or `EE` labels. EE requests are a bit slower on our side as some of us here don't use Premium/Ultimate, so definitely something we could use help with.

Or just find some easy picks in the general issue list as well, I'm sure there's some low-hanging fruit :)

**Pull request merged**